### ISSUE TITLE
Are we using C++14?

### ISSUE THREAD

**Issue by dsroche** [linbox-team/linbox#126]
The most recent commit to master (0bb747ae) has an `enable_if_t`, which is causing me some compilation errors. I believe that helper struct is only part of C++14, but it seems that the default compiler flag in Linbox is g++11.

So what should it be? Personally I would be happy for C++14; the compiler support for clang and g++ seems to be pretty stable and it's not a big change from C++11 as far as I know, mostly just some useful help (such as **auto return types** hooray!). Or if it's only C++11, I guess the `enable_if_t` is a bug.

**Comment by Breush** [comment]
@jgdumas What makes it not to be possible for Sage?
We might be able to use C++14 internally but show a C++11 interface.

**Comment by ClementPernet** [comment]
@Breush : not really, as SageMath uses LinBox as a source code library and compiles it, so we should require SageMath to use all the flags required to compile LinBox, including `--std=c++14`.
I'll raise the question on sage-devel about whether or not this seems reasonable to move to c++14.
